### ISSUE TITLE
Ambiguity when mixing exclusive and normal systems

### ISSUE THREAD

**Issue by thebluefish** [IyesGames/iyes_loopless#15]
It seems I must manually bring in `iyes_loopless::condition::IntoConditionalExclusiveSystem` to support exclusive systems with conditions, however this appears to cause an ambiguity problem with non-exclusive systems if I bring it into the same scope. This example demonstrates my issue:

```rs
use bevy::prelude::*;
use iyes_loopless::prelude::*;
use iyes_loopless::condition::IntoConditionalExclusiveSystem;

fn main() {
    App::new()
        .add_plugins(MinimalPlugins)
        .add_loopless_state(DevState::Loading)
        .add_enter_system(DevState::Loading, setup)
        .add_system(ready
            .run_in_state(DevState::Ready)
            .run_unless_resource_exists::<Thing>()
        )
        .add_system(process
            .run_in_state(DevState::Loading)
            .run_if_resource_exists::<Thing>()
            .at_end()
        )
        .run()
    ;
}

#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
pub enum DevState {
    Loading,
    Ready,
}

#[derive(Debug, Clone, Copy)]
pub struct Thing(pub u32);

fn setup(mut commands: Commands) {
    commands.insert_resource(Thing(1));
}

fn ready(mut commands: Commands) {
    println!("ready");
    commands.insert_resource(Thing(2));
}

fn process(mut world: &mut World) {
    println!("removed {:?}", world.remove_resource::<Thing>());
    world.insert_resource(NextState(DevState::Ready));
}
```
```error[E0034]: multiple applicable items in scope
   --> examples\test.rs:11:14
    |
11  |             .run_in_state(DevState::Loading)
    |              ^^^^^^^^^^^^ multiple `run_in_state` found
    |
    = note: candidate #1 is defined in an impl of the trait `iyes_loopless::condition::IntoConditionalSystem` for the type `S`
    = note: candidate #2 is defined in an impl of the trait `IntoConditionalExclusiveSystem` for the type `S`
note: candidate #3 is defined in the trait `iyes_loopless::condition::ConditionHelpers`
   --> C:\Users\s33n\.cargo\registry\src\github.com-1ecc6299db9ec823\iyes_loopless-0.5.1\src\condition.rs:366:5
    |
366 |     fn run_in_state<T: bevy_ecs::schedule::StateData>(self, state: T) -> Self {
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
help: disambiguate the associated function for candidate #1
    |
10  |         .add_system(iyes_loopless::condition::IntoConditionalSystem::run_in_state(ready, DevState::Loading)
    |                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
help: disambiguate the associated function for candidate #2
    |
10  |         .add_system(IntoConditionalExclusiveSystem::run_in_state(ready, DevState::Loading)
    |                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
help: disambiguate the associated function for candidate #3
    |
10  |         .add_system(iyes_loopless::condition::ConditionHelpers::run_in_state(ready, DevState::Loading)
    |                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

One option I see is to fully qualify the extension for each exclusive system I add. This works, but seems verbose and ugly:
```rs
.add_system(iyes_loopless::condition::IntoConditionalExclusiveSystem::run_in_state(process,DevState::Loading)
    .run_if_resource_exists::<Thing>()
    .at_end()
)
```
Another option I've found is to scope the exclusive trait. This at least preserves the look and feel of the builder API, but still feels quite verbose:
```rs
use bevy::prelude::*;
use iyes_loopless::prelude::*;

fn main() {
    let mut app = App::new();
    app.add_plugins(MinimalPlugins)
        .add_loopless_state(DevState::Loading)
        .add_enter_system(DevState::Loading, setup)
        .add_system(ready
            .run_in_state(DevState::Ready)
            .run_unless_resource_exists::<Thing>()
        )
    ;
    // scope for exclusive systems
    {
        use iyes_loopless::condition::IntoConditionalExclusiveSystem;

        app.add_system(process
            .run_in_state(DevState::Loading)
            .run_if_resource_exists::<Thing>()
            .at_end()
        );
    }
    app.run();
}
```
Any better options?

**Comment by robo-corg** [comment]
Hit the same issue while experimenting with exclusive systems powered by lua (mostly for prototyping). @thebluefish 's fix works pretty good.

I wonder if the helper methods need to be unified to some single trait somehow so dispatching can happen off some marker/wrapper type + the existing normal systems? 

I am happy to have to do an additional `my_system.exclusive_system()` if that would help. Not sure if wrapping exclusive systems in some marker type would let you do dispatch all the `run_in_state` style helper methods from one trait or not but if its possible that seems like a better solution than traits with colliding methods names.

**Comment by Shatur** [comment]
Should be fixed by f51431e0e6e7f8f5c77d26e66394a5e9a999aefe?

**Comment by inodentry** [comment]
Yes
